CHANGELOG.md

```diff
@@ -3,6 +3,76 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.8.0
+
+_Jun 28, 2024_
+
+We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:
+
+- 🛰 Introduce server-side data source for improved server integration in the Data Grid. See [documentation](https://mui.com/x/react-data-grid/server-side-data/) for more details.
+- 📈 Support Date data on the BarChart component
+- 🖱️ Support modifying the expansion trigger on the Tree View components
+- ↕️ Support custom column sort icons on the Data Grid
+- 📝 Add "Overlays" section to the Data Grid documentation
```

Comment:
It's up for debate, but I think we have too many highlights and not all of them are impact-full enough to be here.

Do we usually add highlights for new doc sections?
Same for the `Date` data support on BarChart, it makes it look like a major new feature but the PR is super small (#13471)
I could even drop the custom expansion trigger highlight, they are all super small additions compared with the data source.

Reply:
I'm not too sure what constitutes as impactful enough yet, so happy to take the consensus here.

Reply:
I'm not 100% confident saying which features of other teams is impactful enough.

For eXplore, I'd say nothing this week, on a small release I would keep the "expansion trigger" but it's more to have something to say.

Reply:
I'd agree that not adding super-smaller changes to the highlights would put more emphasis on the comparatively larger feature with a much higher anticipated audience. 👍 
But I am not sure which ones to keep and which ones to remove.

We could also split the data source highlight into multiple lines and add a bit more detail to improve the emphasis. 🤔

Reply:
Totally agree that the data source is worth a code snippet :ok_hand: (usually we use a GIF but it's not really a design feature)

Reply:
> Same for the Date data support on BarChart, it makes it look like a major new feature but the PR is super small

I don't think the PR size matters. I added it in the changelog because it impacts how charts are rendered if you use `Date` object. I theory everybody should be happy about this modification, but I prefer to have a highlight to let user see their is a modification

Reply:
OK if you think the `Date` support is worth highlighting, you clearly know better than me :+1: